modules/appeals_api/app/controllers/appeals_api/metadata_controller.rb

```diff
@@ -41,40 +41,45 @@ def appeals_status
     end
 
     def healthcheck
-      if AppealsApi::HealthChecker.services_are_healthy?
-        render json: healthy_service_response
-      else
-        render json: unhealthy_service_response,
-               status: :service_unavailable
-      end
+      render json: {
+        description: 'Appeals API health check',
+        status: 'UP',
+        time: Time.zone.now.to_formatted_s(:iso8601)
+      }
     end
 
-    private
+    def upstream_healthcheck
+      health_checker = AppealsApi::HealthChecker.new
+      time = Time.zone.now.to_formatted_s(:iso8601)
 
-    def healthy_service_response
-      {
-        data: {
-          id: 'appeals_healthcheck',
-          type: 'appeals_healthcheck',
-          attributes: {
-            healthy: true,
-            date: Time.zone.now.to_formatted_s(:iso8601)
-          }
+      render json: {
+        description: 'Appeals API upstream health check',
+        status: health_checker.services_are_healthy? ? 'UP' : 'DOWN',
+        time: time,
+        details: {
+          name: 'All upstream services',
+          upstreamServices: AppealsApi::HealthChecker::SERVICES.map do |service|
+                              upstream_service_details(service, health_checker, time)
+                            end
         }
-      }.to_json
+      }, status: health_checker.services_are_healthy? ? 200 : 503
     end
 
-    def unhealthy_service_response
+    private
+
+    def upstream_service_details(service_name, health_checker, time)
+      healthy = health_checker.send("#{service_name.snakecase}_is_healthy?")
```

Comment:
you can avoid using metaprogramming  if this line was like `healthy = health_checker.healthy_service?(service_name)`
and then you had the method in health_checker like this:
```
    def healthy_service?(service)
      case service
      when /caseflow/i
        caseflow_is_healthy?
      else
        raise "AppealsApi::HealthChecker doesn't recognize #{service}"
      end
    end
```

Reply:
Thanks for the suggestion, updated!